client/src/site-search/form.tsx

```diff
@@ -0,0 +1,118 @@
+import React from "react";
+import { Link, useSearchParams } from "react-router-dom";
+import LANGUAGES_RAW from "../languages.json";
+
+import { useLocale } from "../hooks";
+import { appendURL } from "./utils";
+
+const LANGUAGES = new Map(
+  Object.entries(LANGUAGES_RAW).map(([locale, data]) => {
+    return [locale.toLowerCase(), data];
+  })
+);
+
+export default function SiteSearchForm() {
+  const locale = useLocale();
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const query = searchParams.get("q");
+  const [newQ, setNewQ] = React.useState("");
+  React.useEffect(() => {
+    setNewQ(query || "");
+  }, [query]);
+
+  return (
+    <form
+      action={`/${locale}/search`}
+      onSubmit={(event) => {
+        event.preventDefault();
+        setSearchParams(appendURL(searchParams, { q: newQ, page: undefined }));
+      }}
+    >
+      <input
+        type="search"
+        name="q"
+        value={newQ}
+        onChange={(event) => {
+          setNewQ(event.target.value);
+        }}
+      />{" "}
+      <button type="submit">Search</button> <AdvancedOptions />
+    </form>
+  );
+}
+
+function AdvancedOptions() {
+  const locale = useLocale();
+  const [searchParams] = useSearchParams();
+
+  const queryLocales = searchParams.getAll("locale");
+
+  return (
+    <div className="advanced-options">
```

Comment:
@schalkneethling Today, there's only 0 or 1 things that can appear inside this div. 
Some day in the near future I hope to add some sort of widget where you can select "tree". E.g. some widget that offers a list of choices like "CSS", "JavaScript", "Graphics", "WebGL", "Learn", etc. And clicking on any of these would append to the URL, for example, `&prefix=web/css`. 
So keep it in mind that this `<div class="advanced-options">` might need to grow vertically.

Reply:
If they are not deprecated or invalidate the document it would be the case to be able to use any element

Reply:
This thing is already done if you have a form with get method and a select with a name, it is the default behavior of the forms